Week02/Assignment/YWang_02.py

```diff
@@ -0,0 +1,87 @@
+# Week 02 Assignment: Fitting and Plotting
+# Fit a linear curve to the data
+# Fit a cubic curve using the SciPy library
+# Find the area underneath the cubic curve over the domain of the data using the tools in SciPy
+# Plot the data, the linear fit, and the cubic fit in Matplotlib. Make sure to give the plot a title and an x and y label. Save this figure and include it in your pull request!
+# Put the area of the curve on the plot somewhere as text using Matplotlib
+# Use the Bayesian information criterion to justify which model (linear or cubic) is preferable. Include this justification in your journal entry for the week.
+
+# import packages
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.stats import linregress
+from scipy.integrate import simps
+from matplotlib.patches import Polygon
+
+
+# input data
+x = np.array([ 1.,1.5,2.,2.5,3.,3.5,4.,4.5,5.,5.5,6.,6.5,7.,7.5,8.,8.5,9.,9.5,10.])
+y = np.array([3.43,4.94,6.45,9.22,6.32,6.11,4.63,8.95,7.8,8.35,11.45,14.71,11.97,12.46,17.42,17.0,15.45,19.15,20.86])
+
+
+# 1 Fit a linear curve to the data
+slope, intercept, r_value, p_value, std_err = linregress(x,y)
+
+
+# 2 Fit a cubic curve using the SciPy library
+coefficients = np.polyfit(x,y,3)
+p = np.poly1d(coefficients)
```

Comment:
This is a legitimate way to fit a polynomial function, but could you go back and fit the function using scipy rather than numpy? Have a look at the `curve_fit` function we discussed.

https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html